### DISCUSSION TITLE
Closing the connection on component destroy

### DISCUSSION THREAD

**Discussion by yshalsager** [capacitor-community/sqlite#580] [Q&A] [answered]
I am using SQLite plugin with Svelte. My understanding is: SQLite is embedded, so if the code has stopped, the connection by definition is closed.

Do I have to close the db connection if I need it all the time the app is working? Or is it closed implicitly when the app is stopped? e.g `onDestroy(async () => await close_database('data'))`

**Comment by jepiqueau** [comment, marked as answer]
@yshalsager i think it is a better usage to close the connection  which will close the database  before closing the app.
